buildSrc/subprojects/performance/performance.gradle.kts

```diff
@@ -3,8 +3,15 @@ apply(plugin = "org.gradle.kotlin.kotlin-dsl")
 dependencies {
     api(project(":integrationTesting"))
     implementation(project(":build"))
+    implementation("org.openmbee.junit:junit-xml-parser:1.0.0")
     implementation("org.codehaus.groovy.modules.http-builder:http-builder:0.7.2") {
         // Xerces on the runtime classpath is breaking some of our doc tasks
         exclude(group = "xerces")
     }
+    if (JavaVersion.current().isJava9Compatible) {
+        // Java 9 throws ClassNotFoundException
+        implementation("javax.activation:activation:1.1.1")
+        // only high version of lombok supports Java 9
+        implementation("org.projectlombok:lombok:1.18.2")
```

Comment:
Why do we need lombok now? IIUC, lombok is (hopefully) needed only at configuration time and not at runtime. Can we rewrite the module descriptor of `junit-xml-parser` so we fix the versions of all the new dependencies we pull in? I really would like to keep the number of third party dependencies to a minimum...

Reply:
Because `junit-xml-parser` depends on an old version of `lombok` which doesn't support Java 9: https://mvnrepository.com/artifact/org.openmbee.junit/junit-xml-parser/1.0.0

What do you mean "rewrite the module descriptor"?

Reply:
Using a component metadata rule (or whatever we are supposed to use now) to change the dependencies of `junit-xml-parser` instead of adding new dependencies which hopefully override the dependencies to `junit-xml-parser`.

What I am saying is that I don't think that we need the lombok dependency at runtime at all. Does the build fail when we exclude lombok?

Reply:
Yes, it fails with the same issue here: https://github.com/rzwitserloot/lombok/issues/1290

Reply:
So that means that lombok now tries to do annotation processing on the build src project? Another reason to remove it. At most, it should be required at runtime.

Reply:
And by removing I mean remove the transitive dependency on Lombok from junit-xml-parser.

Reply:
This dependency declaration works for me with Java 8 and 9:
```
    implementation("org.openmbee.junit:junit-xml-parser:1.0.0") {
        exclude(module = "lombok")
    }
    implementation("javax.activation:activation:1.1.1")
```

Reply:
The component metadata rule version looks like this:

```kotlin
dependencies {
    implementation("org.openmbee.junit:junit-xml-parser:1.0.0")

    components {
        withModule("org.openmbee.junit:junit-xml-parser", JUnitXmlParserRule::class.java)
    }
}

open class JUnitXmlParserRule : ComponentMetadataRule {
    override fun execute(context: ComponentMetadataContext) {
        context.details.allVariants {
            withDependencies {
                removeAll { it.name == "lombok" }
                add("javax.activation:activation:1.1.1")
            }
        }
    }
}
```

Reply:
Thanks, good to know!